# libcss: select-properties [fixes136a]
# 2 files, 166 changed lines
# MacSurf fix ids: fixes136a
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libcss/src/select/properties/s_overflow_wrap.c browser/libcss/src/select/properties/s_overflow_wrap.c
--- upstream/libcss/src/select/properties/s_overflow_wrap.c	1970-01-01 01:00:00.000000000 +0100
+++ browser/libcss/src/select/properties/s_overflow_wrap.c	2026-09-20 23:19:17.250356182 +0100
@@ -0,0 +1,83 @@
+/*
+ * This file is part of LibCSS
+ * Licensed under the MIT License,
+ *                http://www.opensource.org/licenses/mit-license.php
+ * Copyright 2026 MacSurf (fixes136a)
+ */
+
+#include "bytecode/bytecode.h"
+#include "bytecode/opcodes.h"
+#include "select/propset.h"
+#include "select/propget.h"
+#include "utils/utils.h"
+
+#include "select/properties/properties.h"
+#include "select/properties/helpers.h"
+
+css_error css__cascade_overflow_wrap(uint32_t opv, css_style *style,
+		css_select_state *state)
+{
+	uint16_t value = CSS_OVERFLOW_WRAP_INHERIT;
+
+	UNUSED(style);
+
+	if (hasFlagValue(opv) == false) {
+		switch (getValue(opv)) {
+		case CSS_OVERFLOW_WRAP_NORMAL:
+			value = CSS_OVERFLOW_WRAP_NORMAL;
+			break;
+		case CSS_OVERFLOW_WRAP_BREAK_WORD:
+			value = CSS_OVERFLOW_WRAP_BREAK_WORD;
+			break;
+		case CSS_OVERFLOW_WRAP_ANYWHERE:
+			value = CSS_OVERFLOW_WRAP_ANYWHERE;
+			break;
+		}
+	}
+
+	if (css__outranks_existing(getOpcode(opv), isImportant(opv), state,
+			getFlagValue(opv))) {
+		return set_overflow_wrap(state->computed, (uint8_t)value);
+	}
+
+	return CSS_OK;
+}
+
+css_error css__set_overflow_wrap_from_hint(const css_hint *hint,
+		css_computed_style *style)
+{
+	return set_overflow_wrap(style, hint->status);
+}
+
+css_error css__initial_overflow_wrap(css_select_state *state)
+{
+	return set_overflow_wrap(state->computed, CSS_OVERFLOW_WRAP_NORMAL);
+}
+
+css_error css__copy_overflow_wrap(
+		const css_computed_style *from,
+		css_computed_style *to)
+{
+	if (from == to) {
+		return CSS_OK;
+	}
+
+	return set_overflow_wrap(to, get_overflow_wrap(from));
+}
+
+css_error css__compose_overflow_wrap(const css_computed_style *parent,
+		const css_computed_style *child,
+		css_computed_style *result)
+{
+	uint8_t type = get_overflow_wrap(child);
+
+	return css__copy_overflow_wrap(
+			type == CSS_OVERFLOW_WRAP_INHERIT ? parent : child,
+			result);
+}
+
+uint32_t destroy_overflow_wrap(void *bytecode)
+{
+	(void)bytecode;
+	return 0;
+}
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libcss/src/select/properties/s_word_break.c browser/libcss/src/select/properties/s_word_break.c
--- upstream/libcss/src/select/properties/s_word_break.c	1970-01-01 01:00:00.000000000 +0100
+++ browser/libcss/src/select/properties/s_word_break.c	2026-09-20 23:19:17.253415877 +0100
@@ -0,0 +1,83 @@
+/*
+ * This file is part of LibCSS
+ * Licensed under the MIT License,
+ *                http://www.opensource.org/licenses/mit-license.php
+ * Copyright 2026 MacSurf (fixes136a)
+ */
+
+#include "bytecode/bytecode.h"
+#include "bytecode/opcodes.h"
+#include "select/propset.h"
+#include "select/propget.h"
+#include "utils/utils.h"
+
+#include "select/properties/properties.h"
+#include "select/properties/helpers.h"
+
+css_error css__cascade_word_break(uint32_t opv, css_style *style,
+		css_select_state *state)
+{
+	uint16_t value = CSS_WORD_BREAK_INHERIT;
+
+	UNUSED(style);
+
+	if (hasFlagValue(opv) == false) {
+		switch (getValue(opv)) {
+		case CSS_WORD_BREAK_NORMAL:
+			value = CSS_WORD_BREAK_NORMAL;
+			break;
+		case CSS_WORD_BREAK_BREAK_ALL:
+			value = CSS_WORD_BREAK_BREAK_ALL;
+			break;
+		case CSS_WORD_BREAK_KEEP_ALL:
+			value = CSS_WORD_BREAK_KEEP_ALL;
+			break;
+		}
+	}
+
+	if (css__outranks_existing(getOpcode(opv), isImportant(opv), state,
+			getFlagValue(opv))) {
+		return set_word_break(state->computed, (uint8_t)value);
+	}
+
+	return CSS_OK;
+}
+
+css_error css__set_word_break_from_hint(const css_hint *hint,
+		css_computed_style *style)
+{
+	return set_word_break(style, hint->status);
+}
+
+css_error css__initial_word_break(css_select_state *state)
+{
+	return set_word_break(state->computed, CSS_WORD_BREAK_NORMAL);
+}
+
+css_error css__copy_word_break(
+		const css_computed_style *from,
+		css_computed_style *to)
+{
+	if (from == to) {
+		return CSS_OK;
+	}
+
+	return set_word_break(to, get_word_break(from));
+}
+
+css_error css__compose_word_break(const css_computed_style *parent,
+		const css_computed_style *child,
+		css_computed_style *result)
+{
+	uint8_t type = get_word_break(child);
+
+	return css__copy_word_break(
+			type == CSS_WORD_BREAK_INHERIT ? parent : child,
+			result);
+}
+
+uint32_t destroy_word_break(void *bytecode)
+{
+	(void)bytecode;
+	return 0;
+}
